### PR TITLE
Enable on spilling test on left/right semi projection types

### PR DESCRIPTION
**Pull request by xiaoxmeng**
Also fix a spilling document alignment

### CODE REVIEW

File: velox/exec/tests/HashJoinTest.cpp [hunk 1]
```diff
@@ -2683,6 +2683,118 @@ TEST_P(MultiThreadedHashJoinTest, noSpillLevelLimit) {
...
+                 "",
+                 {"t0", "t1", "match"},
+                 core::JoinType::kLeftSemiProject)
+             .planNode();
+
+  HashJoinBuilder(*pool_, duckDbQueryRunner_)
+      .planNode(plan)
+      .referenceQuery(
+          "SELECT t.c0, t.c1, EXISTS (SELECT * FROM u WHERE u.c0 < 0 AND t.c0 = u.c0) FROM t")
+      .checkSpillStats(false)
```
Comment (mbasmanova) [inline velox/exec/tests/HashJoinTest.cpp:2787]: nit: Perhaps, add a comment to explain why we are not checking spilling stats.

File: velox/exec/tests/HashJoinTest.cpp [hunk 2]
```diff
@@ -2683,6 +2683,118 @@ TEST_P(MultiThreadedHashJoinTest, noSpillLevelLimit) {
       .run();
 }
 
+TEST_F(HashJoinTest, semiProject) {
```
Comment (mbasmanova) [inline velox/exec/tests/HashJoinTest.cpp:2686]: Curious, why was it necessary to move the test? It makes it hard to tell what's changed.

**Review by mbasmanova — APPROVED** [review]
@xiaoxmeng Thank you for increasing test coverage.

File: velox/exec/tests/HashJoinTest.cpp (hunk 2 above)
Comment (xiaoxmeng) [reply velox/exec/tests/HashJoinTest.cpp:2686]: Yeah, change from TEST_F to TEST_P to test with different number of drivers.

File: velox/exec/tests/HashJoinTest.cpp (hunk 1 above)
Comment (xiaoxmeng) [reply velox/exec/tests/HashJoinTest.cpp:2787]: Sure. Thanks!

**Comment by xiaoxmeng** [comment]
> @xiaoxmeng Thank you for increasing test coverage.

@mbasmanova  Thanks for the quick review!

**Pull request closed without merging**